tests/common/reboot.py

```diff
@@ -105,6 +105,10 @@ def reboot(duthost, localhost, reboot_type='cold', delay=10, \
     try:
         reboot_ctrl    = reboot_ctrl_dict[reboot_type]
         reboot_command = reboot_ctrl['command'] if reboot_type != REBOOT_TYPE_POWEROFF else None
+        if reboot_type == REBOOT_TYPE_WATCHDOG:
+            res = duthost.command('python -c \"import sonic_platform\"',module_ignore_errors=True)
+            if res['failed']:
+                reboot_command = "python3 -c \"import sonic_platform.platform as P; P.Platform().get_chassis().get_watchdog().arm(5); exit()\""
```

Comment:
Is it possible to templatize the command stored in `reboot_ctrl_dict` like `python{} -c ...` so that you can simply pass 2 or 3 into it here, rather than recreating the command here using Python 3? This method defeats the purpose of storing these commands nicely in a dict.

Reply:
I tried with changing the command definition above but it didn't work with some python library error so I just added this change since python2 case is already covered by the original command.

Reply:
I think you should be able to make it work. Can you paste the Python error? This solution is a bit messy, as the Python 2 command lives in the dict but the Python 3 command is only here. So if any changes are necessary, one needs to know to make changes in both.

Reply:
@jleveque Can you please review again?

Reply:
There's still duplication with the `REBOOT_TYPE_WATCHDOG` entry in `reboot_ctrl_dict `; now it's unused. I think it would be better to either templatize the command in that entry or to rename that entry to `REBOOT_TYPE_WATCHDOG_PY2` and add a new `REBOOT_TYPE_WATCHDOG_PY3` entry and call the appropriate one based on the environment.